scene/animation/animation_mixer.cpp

```diff
@@ -399,12 +399,18 @@ void AnimationMixer::get_animation_list(List<StringName> *p_animations) const {
 	}
 }
 
-Ref<Animation> AnimationMixer::get_animation(const StringName &p_name) const {
-	ERR_FAIL_COND_V_MSG(!animation_set.has(p_name), Ref<Animation>(), vformat("Animation not found: \"%s\".", p_name));
-	const AnimationData &anim_data = animation_set[p_name];
+const Ref<Animation> &AnimationMixer::get_animation(const StringName &p_name) const {
+	static const Ref<Animation> EMPTY_ANIMATION = Ref<Animation>();
+	AHashMap<StringName, AnimationData>::ConstIterator it = animation_set.find(p_name);
+	ERR_FAIL_COND_V_MSG(!it, EMPTY_ANIMATION, vformat("Animation not found: \"%s\".", p_name));
+	const AnimationData &anim_data = it->value;
```

Comment:
Is making this `const` necessary? The calling sites aren't const, so wouldn't any benefit you get from using const here be negated?

Reply:
It's mostly about making code easier to read, so you know the value won't be reassigned after it's assigned once. It's similar to using `const` instead of `let` in JavaScript in that sense.

Reply:
The question isn't about the local variable. It is returning a `const Ref<Animation>` which requires us to create a new wrapper function `_get_animation()` below which does nothing but convert the `const Ref<Animation>` to a non const `Ref<Animation>`